torchvision/transforms/functional_tensor.py

```diff
@@ -466,8 +466,18 @@ def resize(
     # Define align_corners to avoid warnings
     align_corners = False if interpolation in ["bilinear", "bicubic"] else None
 
+    # This is a perf hack to avoid <pytorch-issue>
+    # We are transforming (1, 1, H, W) into (1, 2, H, W) to force to take channels_first path
```

Comment:
What if the input is `(N, 1, H, W)`, do you still need to do this hack? Will your code work in this scenario? What's the performance gain for batch of say 32?

Reply:
It gives ~4x speedup if using batch=32:
```


Timestamp: 20220928-181325
Torch version: 1.13.0.dev20220906+cu113
Torchvision version: 0.14.0a0
Num threads: 1

[--------------------------------- Mask Resize measurements --------------------------------]
                                       |  Original (slow) mask 2d  |  Hacked (faster) mask 2d
1 threads: ----------------------------------------------------------------------------------
      ([32, 1, 500, 500]), 500 -> 128  |            5.5            |            1.6          

Times are in milliseconds (ms).
```